modules/tide_site_simple_sitemap/src/Plugin/simple_sitemap/SitemapGenerator/TideDefaultSitemapGenerator.php

```diff
@@ -21,14 +21,18 @@ class TideDefaultSitemapGenerator extends DefaultSitemapGenerator {
    */
   public function generate(array $links) {
     parent::generate($links);
-    $highest_id = $this->db->query('SELECT MAX(id) FROM {simple_sitemap_site}')
+    $highest_id = $this->db->query('SELECT MAX(id) FROM {simple_sitemap}')
       ->fetchField();
-    $highest_delta = $this->db->query('SELECT MAX(delta) FROM {simple_sitemap_site} WHERE type = :type AND status = :status', [
+    $highest_delta = $this->db->query('SELECT MAX(delta) FROM {simple_sitemap} WHERE type = :type AND status = :status', [
       ':type' => $this->sitemapVariant,
-      ':status' => 0,
+      ':status' => 1,
     ])
       ->fetchField();
-    $this->db->truncate('simple_sitemap_site')->execute();
+
+    $this->db->delete('simple_sitemap_site')
```

Comment:
This line would cause the same issue as the truncate statement. You only need to delete the existing row with the same ID(s) of the row you are about to insert so that database error would not occur.

Reply:
thanks, @sonnykt , that is correct. I have removed the deletion part.